net/adguardhome/files/adguardhome.init

```diff
@@ -9,15 +11,51 @@ START=21
 # stops before networking stops
 STOP=89
 
+handle_extra_mount() {
+  local mount="$1"
+
+  append EXTRA_MOUNT "$mount"
+}
+
 start_service() {
   config_load adguardhome
   config_get WORK_DIR config workdir
+  config_get_bool jail config jail 1
+
+  EXTRA_MOUNT=
+
+  config_list_foreach config jail_mount handle_extra_mount
+
+  if [ ! -d "$CONFIG_DIR" ]; then
+    mkdir -p "$CONFIG_DIR"
+    chown adguardhome:adguardhome "$CONFIG_DIR"
+  fi
 
-  [ -d "$WORK_DIR" ] || mkdir -m 0755 -p "$WORK_DIR"
+  if [ ! -d "$WORK_DIR" ]; then
+    mkdir -p "$WORK_DIR"
+    chown adguardhome:adguardhome "$WORK_DIR"
+  fi
 
   procd_open_instance
-  procd_set_param command "$PROG" -c /etc/adguardhome.yaml -w "$WORK_DIR" --no-check-update
+  procd_set_param command "$PROG" -c "$CONFIG_FILE" -w "$WORK_DIR" --no-check-update
   procd_set_param stdout 1
   procd_set_param stderr 1
+
+  if [ "$jail" -ne 0 ]; then
```

Comment:
this appears prone to the 'umdns ujail not installed bug'...

until a usable library exists, you are going to have to check for jail capability and consider what to do if it's not available...

Reply:
Yea, fixed with a `-x /sbin/ujail` check.